devito/types/grid.py

```diff
@@ -182,6 +182,19 @@ def subdomains(self):
         """The SubDomains defined in this Grid."""
         return {i.name: i for i in self._subdomains}
 
+    def add_subdomains(self, subdomain_dict):
```

Comment:
``Grid`` should be immutable, so no add-like methods. One should create a Grid and pass the pre-created SubDomains through the constructor

Reply:
The issue was that `class AbstractModel` creates a grid with some required underlying subdomains which the user isn't directly exposed to. You can then pass a grid to `class Model` but this breaks things as it won't have the correct subdomains. I'll therefore have to these classes a bit.